openedx/core/djangoapps/content_libraries/libraries_index.py

```diff
@@ -0,0 +1,302 @@
+                                        'query': text_search_normalised,
+                                        "fields": ["content.*"],
+                                        "minimum_should_match": "100%",
+                                    },
+                                },
+                                # Add a special wildcard search for id, as it contains a ":" character which is filtered out
+                                # in query_string
+                                {
+                                    'wildcard': {
+                                        'id': {
+                                            'value': '*{}*'.format(text_search),
+                                        }
+                                    },
+                                },
+                            ],
+                        },
+                    },
+                    'filter': {
+                        'bool': {
+                            'must': terms
+                        }
+                    }
+                },
+            },
+        }
+
+
+class ContentLibraryIndexer(SearchIndexerBase):
+    """
+    Class to perform indexing for blockstore-based content libraries
+    """
+
+    INDEX_NAME = "content_library_index"
+    ENABLE_INDEXING_KEY = "ENABLE_CONTENT_LIBRARY_INDEX"
+    DOCUMENT_TYPE = "content_library"
+
+    @classmethod
+    def get_item_definition(cls, library_key):
+        from openedx.core.djangoapps.content_libraries.api import DRAFT_NAME
+
+        ref = ContentLibrary.objects.get_by_key(library_key)
+        lib_bundle = LibraryBundle(library_key, ref.bundle_uuid, draft_name=DRAFT_NAME)
+        num_blocks = len(lib_bundle.get_top_level_usages())
+        last_published = lib_bundle.get_last_published_time()
+        last_published_str = None
+        if last_published:
+            last_published_str = last_published.strftime('%Y-%m-%dT%H:%M:%SZ')
+        (has_unpublished_changes, has_unpublished_deletes) = lib_bundle.has_changes()
+
+        bundle_metadata = get_bundle(ref.bundle_uuid)
+
+        return {
+            "id": str(library_key),
+            "uuid": str(bundle_metadata.uuid),
+            "title": bundle_metadata.title,
+            "description": bundle_metadata.description,
+            "num_blocks": num_blocks,
+            "version": bundle_metadata.latest_version,
+            "last_published": last_published_str,
+            "has_unpublished_changes": has_unpublished_changes,
+            "has_unpublished_deletes": has_unpublished_deletes,
+            # only 'content' field is analyzed by elastisearch, and allows text-search
+            "content": {
+                "id": str(library_key),
+                "title": bundle_metadata.title,
+                "description": bundle_metadata.description,
+            },
+        }
+
+
+class LibraryBlockIndexer(SearchIndexerBase):
+    """
+    Class to perform indexing for blockstore-based content libraries
```

Comment:
Nit: maybe the doc should be at least a little different from the ContentLibraryIndexer's one.